doc/002-local-dev.md

```diff
@@ -1,184 +1,207 @@
+### `ziti` CLI
 
-#### Build the Container Image
+```bash
+# build the binary without installing in GOPATH
+go build -o ./build/ziti ./ziti/cmd/ziti/
+# execute the binary
+./build/ziti version
+```
 
-You only need to build the container image once unless you change the Dockerfile or `./linux-build.sh` (the container's entrypoint).
+### `ziti-controller`
 
 ```bash
-# find the latest Go distribution's semver
-LATEST_GOLANG=$(curl -sSfL "https://go.dev/VERSION?m=text" | /bin/grep -Po '^go(\s+)?\K\d+\.\d+\.\d+$')
-# build a container image named "zitibuilder" with the Dockerfile in the top-level of this repo
-docker build \
-    --tag=zitibuilder \
-    --file=Dockerfile.linux-build \
-    --build-arg latest_golang=${LATEST_GOLANG} \
-    --build-arg uid=$UID \
-    --build-arg gid=$GID .
+# build the binary without installing in GOPATH
+go build -o ./build/ziti-controller ./ziti-controller/
+# execute the binary
+./build/ziti-controller version
 ```
 
-#### Run the Container to Build Executables for the Desired Architectures
-
-Executing the following `docker run` command will:
-1. Mount the top-level of this repo on the container's `/mnt`
-2. Run `./linux-build.sh ${@}` inside the container
-3. Deposit built executables in `./release`
+### `ziti-router`
 
 ```bash
-# build for all three architectures: amd64 arm arm64
-docker run \
-    --rm \
-    --name=zitibuilder \
-    --volume=$PWD:/mnt \
-    zitibuilder
-
-# build only amd64 
-docker run \
-    --rm \
-    --name=zitibuilder \
-    --volume=$PWD:/mnt \
-    zitibuilder \
-        amd64
+# build the binary without installing in GOPATH
+go build -o ./build/ziti-router ./ziti-router/
+# execute the binary
+./build/ziti-router version
 ```
 
-You will find the built artifacts in `./release`.
+## Crossbuilds
+
+When you push to your repo fork then GitHub Actions will automatically crossbuild for several OSs and CPU architectures. You'll then be able to download the built artifacts from the GitHub UI. The easiest way to crossbuild the Linux exectuables locally is to build and run the crossbuild container. Please refer to [the crossbuild container README](../Dockerfile.linux-build.README) for those steps. For hints on crossbuilding for MacOS and Windows see [the main GitHub Actions workflow](../.github/workflows/main.yml) which defines the steps that are run when you push to GitHub.
+
+## Run a Local Ziti Stack
+
+Let's get a local Ziti stack up and running now that you have built and installed all the Ziti apps in this repo.
+
+### Initialize the Shell Environment
+
+You'll need to export two environment variables that are employed by the included controller and router configuration YAML files. These are needed in each shell where you'll run a controller or router process. You could manage them all as background processes in a single shell, but it's easier to use separate shells for each if only to have a dedicated view for the log messages of each process.
```

Comment:
this is all bash-centric and i'd personally prefer to just remove all this too. Keep ZITI_SOURCE and ZITI_DATA hidden for now imo.

Reply:
We can do that. It means adding a new controller config that doesn't use these two env vars. It might also mean a new .gitignore statement to allow `ZITI_DATA` to be a relative path inside this repo's checked-out copy.

Reply:
We decided to generate the controller and router configs with `ziti` CLI instead of using the checked-in configs